src/pathfinder/src/main.cc

```diff
@@ -26,41 +28,132 @@ struct CtlMsg
 class Pathfinder
 {
   std::vector<float> _angles;
+  std::vector<float> _filtered;
+#define lidar_steering
 
 public:
   Pathfinder() {}
 
   CtlMsg update_frame(sensor_msgs::LaserScan const& frame)
   {
-    if ((_angles.size() == 0) || (_angles[0] != frame.angle_min))
+    float _ranges = frame.ranges;
+    float _angle_min = frame.angle_min;
+    float _angle_max = frame.angle_max;
+    float _angle_increment = frame.angle_increment;
+    float _range_max = frame.range_max;
+    float _range_min = frame.range_min;
+
+    if ((_angles.size() == 0) || (_angles[0] != _angle_min[0]))
+    {
+      _angles.push_back(_angle_min[0]);
+      for (size_t i = 1; i < _ranges.size(); i++)
+        _angles.push_back(_angles.back() + _angle_increment[0]);
+    }
+
+    assert(_angles.size() == _ranges.size());
+
+#ifdef lidar_steering
+    for (size_t i = 0; i < _angles.size(); i++)
+    {
+      _filtered.push_back(_ranges[i]);
+    }
+
+    // Safety radius for car
+    float safety = 0.500 / 2.0;
+
+    for (int i = 0; i < (int)(_angles.size()); i++)
+    {
+      float _r = _ranges[i];
+
+      // Ensures distant points are filtered
+      if (_r > 10)
+      {
+        _r = 10.0;
+      }
+
+      float t = atan2(safety, _r);
+
+      int d_idx = (int)(t / _angle_increment[0]) * 2;
+      int lower = std::max(0, i - d_idx);
+      int upper = std::min((int)(_angles.size()), i + d_idx);
+
+      for (int j = lower; j < upper; j++)
+      {
+        if (_filtered[i] > _ranges[j])
+          _filtered[i] = _ranges[j];
+      }
+    }
+
+    // Selection process
+    float max_filt_range = {0};
+    int path = {0};
+    for (size_t i = 0; i < _filtered.size(); i++)
+    {
+      if (_filtered[i] > max_filt_range)
+      {
+        max_filt_range = _filtered[i];
+        path = i;
+      }
+    }
+
+    // _All the path index variables are the selection
+    printf(
+      "index \t angle \t\t range \t filtered \n%2d \t %.3f \t %.3f \t %.3f\n",
+      path,
+      _angles[path],
+      _ranges[path],
+      _filtered[path]);
+
+    int8_t wheel_angle = round(_angles[path] * 57.30);
```

Comment:
Is `57.30` the conversion factor from radians to angles? If so there should either  be a comment explaining it or, even better, a named constexpr:
```suggestion
    static constexpr float rad_to_deg = 180.0 / M_PI;
    auto wheel_angle = (int8_t)round(_angles[path] * rad_to_deg);
```